src/PointwiseFunctions/AnalyticSolutions/GrMhd/BondiMichelMHD.hpp

```diff
@@ -0,0 +1,296 @@
+    return variables(
+        x, tmpl::list<Tag>{},
+        IntermediateVars<DataType>{rest_mass_density_at_infinity_,
+                                   mass_accretion_rate_over_four_pi_, mass_,
+                                   polytropic_constant_, polytropic_exponent_,
+                                   bernoulli_constant_squared_minus_one_, x});
+  }
+
+  // @{
+  /// Retrieve hydro variable at `(x)`
+  template <typename DataType>
+  auto variables(const tnsr::I<DataType, 3>& x,
+                 tmpl::list<hydro::Tags::RestMassDensity<DataType>> /*meta*/,
+                 const IntermediateVars<DataType>& vars) const noexcept
+      -> tuples::TaggedTuple<hydro::Tags::RestMassDensity<DataType>>;
+
+  template <typename DataType>
+  auto variables(
+      const tnsr::I<DataType, 3>& x,
+      tmpl::list<hydro::Tags::SpecificInternalEnergy<DataType>> /*meta*/,
+      const IntermediateVars<DataType>& vars) const noexcept
+      -> tuples::TaggedTuple<hydro::Tags::SpecificInternalEnergy<DataType>>;
+
+  template <typename DataType>
+  auto variables(const tnsr::I<DataType, 3>& x,
+                 tmpl::list<hydro::Tags::Pressure<DataType>> /*meta*/,
+                 const IntermediateVars<DataType>& vars) const noexcept
+      -> tuples::TaggedTuple<hydro::Tags::Pressure<DataType>>;
+
+  template <typename DataType>
+  auto variables(
+      const tnsr::I<DataType, 3>& x,
+      tmpl::list<
+          hydro::Tags::SpatialVelocity<DataType, 3, Frame::Inertial>> /*meta*/,
+      const IntermediateVars<DataType>& vars) const noexcept
+      -> tuples::TaggedTuple<
+          hydro::Tags::SpatialVelocity<DataType, 3, Frame::Inertial>>;
+
+  template <typename DataType>
+  auto variables(
+      const tnsr::I<DataType, 3>& x,
+      tmpl::list<
+          hydro::Tags::MagneticField<DataType, 3, Frame::Inertial>> /*meta*/,
+      const IntermediateVars<DataType>& vars) const noexcept
+      -> tuples::TaggedTuple<
+          hydro::Tags::MagneticField<DataType, 3, Frame::Inertial>>;
+
+  template <typename DataType>
+  auto variables(
+      const tnsr::I<DataType, 3>& x,
+      tmpl::list<hydro::Tags::DivergenceCleaningField<DataType>> /*meta*/,
+      const IntermediateVars<DataType>& vars) const noexcept
+      -> tuples::TaggedTuple<hydro::Tags::DivergenceCleaningField<DataType>>;
+
+  template <typename DataType>
+  auto variables(const tnsr::I<DataType, 3>& x,
+                 tmpl::list<hydro::Tags::LorentzFactor<DataType>> /*meta*/,
+                 const IntermediateVars<DataType>& vars) const noexcept
+      -> tuples::TaggedTuple<hydro::Tags::LorentzFactor<DataType>>;
+
+  template <typename DataType>
+  auto variables(const tnsr::I<DataType, 3>& x,
+                 tmpl::list<hydro::Tags::SpecificEnthalpy<DataType>> /*meta*/,
+                 const IntermediateVars<DataType>& vars) const noexcept
+      -> tuples::TaggedTuple<hydro::Tags::SpecificEnthalpy<DataType>>;
+  // @}
+ private:
```

Comment:
Don't mix access specifiers unless absolutely necessary for some forward declarations.